Create a documentation page for `torch.ao.quantization.QConfig`

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66222 Quantization docs: add pages for Numeric Suite (Eager and FX)
* #66210 Quantization documentation: move backend section down
* #66201 Quantization docs: rewrite API reference to be more automated
* #66198 Quantization docs: consilidate all API references on a single page
* **#66129 Create a documentation page for `torch.ao.quantization.QConfig`**
* #66125 Create separate documentation pages for quantization observers and fake_quants
* #66122 Create a documentation page for FX graph mode quantization APIs

Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

vkuzo committed:
Create a documentation page for `torch.ao.quantization.QConfig`

Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/qconfig.py:
```diff
@@ -225,6 +293,9 @@ def configure_constructor_to_put_obs_on_module_device(original_constructor):
 
 
 def qconfig_equals(q1: QConfigAny, q2: QConfigAny):
+    """
```
nit: should we define the rules about when qconfigs are equal here as well?

jerryzh168 commented on docs/source/torch.quantization.qconfig.rst:
```diff
@@ -0,0 +1,8 @@
+.. _torch_quantization_qconfig:
```
should we rename this file to torch.ao.quantization.qconfig.rst?

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447614).

vkuzo committed:
Update on "Create a documentation page for `torch.ao.quantization.QConfig`"


Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for `torch.ao.quantization.QConfig`"


Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for `torch.ao.quantization.QConfig`"


Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for `torch.ao.quantization.QConfig`"


Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447614).
